From 2fe898d966aaffb9322f6f6f489e01d3d2f02696 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Mon, 3 Dec 2012 14:56:15 +0100
Subject: [PATCH] Load all algorithms when decrypting stronly-encrypted keys
 in tests

Some strong hashes are not loaded by default in OpenSSL 0.9.8.
OpenSSL_add_all_algorithms() makes them available.
---
 t/local/05_passwd_cb.t | 1 +
 t/local/38_priv-key.t  | 1 +
 2 files changed, 2 insertions(+)

diff --git a/t/local/05_passwd_cb.t b/t/local/05_passwd_cb.t
index 9b3bba2..a91cab5 100644
--- a/t/local/05_passwd_cb.t
+++ b/t/local/05_passwd_cb.t
@@ -9,6 +9,7 @@ use Net::SSLeay;
 Net::SSLeay::randomize();
 Net::SSLeay::load_error_strings();
 Net::SSLeay::add_ssl_algorithms();
+Net::SSLeay::OpenSSL_add_all_algorithms();
 
 my $key_pem = File::Spec->catfile('t', 'data', 'key.pem.e');
 my $key_password = 'secret';
diff --git a/t/local/38_priv-key.t b/t/local/38_priv-key.t
index 87a36b4..a80dbd1 100755
--- a/t/local/38_priv-key.t
+++ b/t/local/38_priv-key.t
@@ -10,6 +10,7 @@ Net::SSLeay::randomize();
 Net::SSLeay::load_error_strings();
 Net::SSLeay::ERR_load_crypto_strings();
 Net::SSLeay::SSLeay_add_ssl_algorithms();
+Net::SSLeay::OpenSSL_add_all_algorithms();
 
 my $key_pem_encrypted       = File::Spec->catfile('t', 'data', 'test_CA1.encrypted_key.pem');
 my $key_pem                 = File::Spec->catfile('t', 'data', 'test_CA1.key.pem');
-- 
1.7.11.7

